File: testClients/testClient.go
package main

import (
	pb "leap/protoBuf"
	"fmt"
	"github.com/golang/protobuf/proto"
	"net"
	"leap/coordinator"
)

func main() {
	testPatient := pb.Patient{Fname: "Han",
		Lname:  "Solo",
		Email:  "hsolo@gmail.com",
		Age:    29,
		Gender: pb.Patient_MALE,
		Weight: 80.0,
		Height: 180}

	out, err := proto.Marshal(&testPatient)
	if err != nil {
		fmt.Println(err)
	}


	conn, err := net.Dial("tcp", coordinator.CLOUD_HOST_PORT)
	if err != nil {
		fmt.Println(err)
	}
	_, err = conn.Write(out)
	if err != nil {
		fmt.Println(err)
	}
	response := make([]byte, 1024)
	n, err := conn.Read(response)
	msg := string(response[:n])
	fmt.Println("This is the response: ", msg)
}

File: cloudAlgos/testClient.py
import socket
import patient_pb2

def createPatient():
    patient = patient_pb2.Patient()
    patient.fname  = "Han"
    patient.lname  = "Solo"
    patient.email  = "hansolo.gmail.com"
    patient.age    = 29
    patient.gender = patient_pb2.Patient.MALE
    patient.weight = 80.0
    patient.height = 180
    return patient


if __name__ == "__main__":
    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    s.connect(('localhost', 8001))
    patient = createPatient()
    s.sendall(patient.SerializeToString())
    data = s.recv(1024)
    print "This is the response: " + repr(data)
    s.close()